### PR TITLE
Support multiple commands on multiple URLs in exechealthz

### LINKED ISSUE #1735: Check multiple commands on multiple URLs in Exechealthz
Refer kubernetes/kubernetes#30633.

Would it be possible to support checking multiple commands on multiple URLs in one Exechealthz process? I am going to run two Exechealthz processes within one pod but consider it would be better if Exechealthz supports this intuitively.

@luxas

### PR DESCRIPTION
**Pull request by MrHohn**
Fixes #1735.

Adding support for multiple commands on URLs for [exec-healthz](https://github.com/kubernetes/contrib/tree/master/exec-healthz).

Example as below:

```
$ ./exechealthz --cmd="ls /tmp/test1" --url="/healthz1" --cmd="ls /tmp/test2" --url="/healthz2"
```

The --url flag indicates the path healthz server needs to serve on.
Notes: Number of cmds and URLs have to be the same (if more than 1). URL need to start with "/". URLs and cmds match up based on their orders (first URL to first cmd).

The previous behavior remains the same (as the previous examples in README).

@thockin @bprashanth @luxas

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/contrib/1739)

### CODE REVIEW

**Comment by thockin** [comment]
I don't know this sub-repo - any opportunity for tests?

**Comment by MrHohn** [comment]
Seems like the only way is to add unit tests under the same folder.

Is it Okay to run the main function in a go routine while using another go routine to verify?

**Comment by thockin** [comment]
no really.  I would try to test this by injecting a pkg/util/exec.Interface
which is a fake, and verifying that it gets called.  I don't want to derail
tooo much - it's all about confidence in the code.  You should not be stuck
doing tests for everything you touch, but if you CAN that is great.

**Comment by bprashanth** [comment]
Tests sound like a good idea. I didn't write them initially because I thought it was borderline complexity, but adding more probably pushes it over the edge. 

You can give the `execworker` an interface to exec and a fakeClock, then trigger the clock and confirm the exec hits. Both those packages are available in main kuberepo (https://github.com/kubernetes/kubernetes/tree/master/pkg/util), but require godeps, which this project has thus far steered clear from. I leave it to you, maybe it's easier to add a godep, or just define your own fake.

**Comment by MrHohn** [comment]
@thockin @bprashanth Thanks for the detailed instructions. I tried `exec` and `clock` packages and added tests mainly for the execWorker.

Not sure to also test the web server or not. I consider holding a real server that occupy a port may introduce uncertainty and flaky result.

cc @matchstick

**Comment by MrHohn** [comment]
Sorry, there is a mistake in exechealthz_test.go. I misunderstood the usage of wait.Poll(). Will fix it soon.

**Comment by MrHohn** [comment]
Fixed.

**Comment by test-foxish** [comment]
recomputing cla status...

File: exec-healthz/exechealthz.go
Comment (thockin) [inline exec-healthz/exechealthz.go]: having a default cmd is not that useful.  I won't block this PR on it, but if anything else needs to be fixed, I would just remove this and demand at least one --cmd

File: exec-healthz/exechealthz.go
Comment (thockin) [inline exec-healthz/exechealthz.go]: rename `h` to `ew` or `worker`

**Review by thockin — APPROVED** [review]
LGTM.  This is a good PR.

**Comment by thockin** [comment]
Now you need to tag this and push it.  I might also try to convince you to move this to github.com/thockin/go-build-template

**Pull request merged**
